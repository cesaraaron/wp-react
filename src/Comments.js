import React from 'react'
import PropTypes from 'prop-types'
import FetchContainer from './components/FetchContainer'
import { connect } from 'react-redux'
import { getErrorMessage, getIsFetching, getCommentsForPost } from './reducers'
import * as types from './actions/types'
import { fetchCommentsByPostId } from './actions'

export const Comment = ({ author_name, content }) => (
  <div>
    <div>{author_name}</div>
    <p>{content.rendered}</p>
  </div>
)

Comment.propTypes = {
  author_name: PropTypes.string.isRequired,
  content: PropTypes.object.isRequired
}

export const CommentList = ({ comments }) =>
  comments
    .filter(c => c.status === 'approved')
    .map(c => <Comment {...c} key={c.id} />)

CommentList.propTypes = {
  comments: PropTypes.array.isRequired
}

const CommentListContainer = ({
  fetchCommentsByPostId,
  data,
  postId,
  ...rest
}) => (
  <FetchContainer
    hasData={data.length > 0}
    onMount={() => fetchCommentsByPostId(postId)}
    render={() => <CommentList comments={data} />}
    {...rest}
  />
)

CommentListContainer.propTypes = {
  fetchCommentsByPostId: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  postId: PropTypes.number.isRequired
}

const mapStateToProps = (state, ownProps) => {
  const { postId } = ownProps
  return {
    postId,
    data: getCommentsForPost(state, postId),
    errorMessage: getErrorMessage(state, types.comments),
    isFetching: getIsFetching(state, types.comments)
  }
}

export default connect(mapStateToProps, { fetchCommentsByPostId })(
  CommentListContainer
)
