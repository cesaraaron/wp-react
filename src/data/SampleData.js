export const posts = [
  {
    id: 1,
    date: '2017-09-15T17:42:48',
    date_gmt: '2017-09-15T17:42:48',
    guid: { rendered: 'http://localhost/?p=1' },
    modified: '2017-09-15T17:42:48',
    modified_gmt: '2017-09-15T17:42:48',
    slug: 'hello-world',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/hello-world',
    title: { rendered: 'Hello world!' },
    content: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    author: 1,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [1],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/1'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 1031,
    date: '2013-03-15T17:23:27',
    date_gmt: '2013-03-15T22:23:27',
    guid: { rendered: 'http://wptest.io/demo/?p=1031' },
    modified: '2013-03-15T17:23:27',
    modified_gmt: '2013-03-15T22:23:27',
    slug: 'tiled-gallery',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/tiled-gallery',
    title: { rendered: 'Tiled Gallery' },
    content: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery.</p>\n<p>You can install <a title="Jetpack for WordPress" href="http://jetpack.me/" target="_blank">Jetpack</a> or <a title="Slim Jetpack" href="http://wordpress.org/extend/plugins/slimjetpack/" target="_blank">Slim Jetpack</a> to test it out.</p>\n\n\t\t<style type=\'text/css\'>\n\t\t\t#gallery-1 {\n\t\t\t\tmargin: auto;\n\t\t\t}\n\t\t\t#gallery-1 .gallery-item {\n\t\t\t\tfloat: left;\n\t\t\t\tmargin-top: 10px;\n\t\t\t\ttext-align: center;\n\t\t\t\twidth: 25%;\n\t\t\t}\n\t\t\t#gallery-1 img {\n\t\t\t\tborder: 2px solid #cfcfcf;\n\t\t\t}\n\t\t\t#gallery-1 .gallery-caption {\n\t\t\t\tmargin-left: 0;\n\t\t\t}\n\t\t\t/* see gallery_shortcode() in wp-includes/media.php */\n\t\t</style>\n\t\t<div id=\'gallery-1\' class=\'gallery galleryid-1031 gallery-columns-4 gallery-size-thumbnail\'><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/fight-club\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/fight-club-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Fight Club" srcset="http://localhost:8080/wp-content/uploads/2013/03/fight-club-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/fight-club-1200x1200.jpg 1200w, http://localhost:8080/wp-content/uploads/2013/03/fight-club-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/the-dark-knight-rises\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="The Dark Knight Rises" srcset="http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/captain-america\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/captain-america-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Captain America" srcset="http://localhost:8080/wp-content/uploads/2013/03/captain-america-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/captain-america-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/man-of-steel\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Man Of Steel" srcset="http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><br style="clear: both" /><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/spider-man\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/spider-man-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="The Amazing Spider Man" srcset="http://localhost:8080/wp-content/uploads/2013/03/spider-man-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/spider-man-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/ironman-2\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/ironman-2-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Iron Man 2" srcset="http://localhost:8080/wp-content/uploads/2013/03/ironman-2-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/ironman-2-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl>\n\t\t\t<br style=\'clear: both\' />\n\t\t</div>\n\n<p>This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery. You can install Jetpack or Slim Jetpack to test it out. This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    author: 6,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [15, 16, 17],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1031' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/6'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1031'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1031/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1031' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1031'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1031'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]
