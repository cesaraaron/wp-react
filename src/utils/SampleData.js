export const posts = [
  {
    id: 1,
    date: '2017-09-15T17:42:48',
    date_gmt: '2017-09-15T17:42:48',
    guid: { rendered: 'http://localhost/?p=1' },
    modified: '2017-09-15T17:42:48',
    modified_gmt: '2017-09-15T17:42:48',
    slug: 'hello-world',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/hello-world',
    title: { rendered: 'Hello world!' },
    content: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    author: 1,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [1],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/1'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 1031,
    date: '2013-03-15T17:23:27',
    date_gmt: '2013-03-15T22:23:27',
    guid: { rendered: 'http://wptest.io/demo/?p=1031' },
    modified: '2013-03-15T17:23:27',
    modified_gmt: '2013-03-15T22:23:27',
    slug: 'tiled-gallery',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/tiled-gallery',
    title: { rendered: 'Tiled Gallery' },
    content: {
      rendered: '<p>titled gallery content</p>',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery. You can install Jetpack or Slim Jetpack to test it out. This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    author: 6,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [15, 16, 17],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1031' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/6'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1031'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1031/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1031' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1031'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1031'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]

export const comments = [
  {
    id: 2,
    post: 1077,
    parent: 0,
    author: 0,
    author_name: 'Chris Ames',
    author_url: 'http://chrisam.es/',
    date: '2013-03-15T18:16:59',
    date_gmt: '2013-03-15T23:16:59',
    content: {
      rendered:
        '<p>Ello! Pretend you&#8217;re reading this comment with an English accent.</p>\n'
    },
    link: 'http://wptest.io/demo/page-comments/#comment-2',
    status: 'approved',
    type: 'comment',
    author_avatar_urls: {
      '24':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=24&d=retro&r=g',
      '48':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=48&d=retro&r=g',
      '96':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=96&d=retro&r=g'
    },
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments/2' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments' }],
      up: [
        {
          embeddable: true,
          post_type: 'page',
          href: 'http://wptest.io/demo/wp-json/wp/v2/pages/1077'
        }
      ],
      children: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/comments?parent=2' }
      ]
    }
  },
  {
    id: 31,
    post: 877,
    parent: 0,
    author: 0,
    author_name: 'Jared Erickson',
    author_url: 'http://jarederickson.com/',
    date: '2013-03-14T13:07:19',
    date_gmt: '2013-03-14T18:07:19',
    content: {
      rendered:
        '<p>Non-breaking text is my favorite!</p>\n<p>Super/Duper/Long/NonBreaking/Path/Name/To/A/File/That/Is/Way/Deep/Down/In/Some/Mysterious/Remote/Desolate/Part/Of/The/Operating/System/To/A/File/That/Just/So/Happens/To/Be/Strangely/Named/Supercalifragilisticexpialidocious.txt</p>\n'
    },
    link: 'http://wptest.io/demo/2013/01/05/non-breaking-text/#comment-31',
    status: 'approved',
    type: 'comment',
    author_avatar_urls: {
      '24':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=24&d=retro&r=g',
      '48':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=48&d=retro&r=g',
      '96':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=96&d=retro&r=g'
    },
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments/31' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments' }],
      up: [
        {
          embeddable: true,
          post_type: 'post',
          href: 'http://wptest.io/demo/wp-json/wp/v2/posts/877'
        }
      ],
      children: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/comments?parent=31' }
      ]
    }
  }
]

export const categories = [
  {
    id: 2,
    count: 3,
    description: 'Posts in this category test image and text alignment.',
    link: 'http://wptest.io/demo/category/alignment/',
    name: 'Alignment',
    slug: 'alignment',
    taxonomy: 'category',
    parent: 0,
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories/2' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories' }],
      about: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/taxonomies/category' }
      ],
      'wp:post_type': [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/posts?categories=2' }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 3,
    count: 2,
    description: 'Posts in this category test the aside post format.',
    link: 'http://wptest.io/demo/category/post-format-aside/',
    name: 'Aside',
    slug: 'post-format-aside',
    taxonomy: 'category',
    parent: 0,
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories/3' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories' }],
      about: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/taxonomies/category' }
      ],
      'wp:post_type': [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/posts?categories=3' }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]
