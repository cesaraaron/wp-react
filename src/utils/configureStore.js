import rootReducer from '../reducers'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'

export default () => {
  const logger = createLogger()

  const store = createStore(rootReducer, applyMiddleware(thunk, logger))

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers')
      store.replaceReducer(nextRootReducer)
    })
  }

  return store
}
