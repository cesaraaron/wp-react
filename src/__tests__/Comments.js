import React from 'react'
import renderer from 'react-test-renderer'
import { Comment, CommentList } from '../Comments'
import { comments } from '../utils/SampleData'

it('should render a comment', () => {
  const component = renderer.create(<Comment {...comments[0]} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render a list of comments', () => {
  const component = renderer.create(<CommentList data={comments} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
