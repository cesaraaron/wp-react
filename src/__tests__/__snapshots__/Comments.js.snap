// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a comment 1`] = `
<div>
  <div>
    Chris Ames
  </div>
  <p>
    &lt;p&gt;Ello! Pretend you&#8217;re reading this comment with an English accent.&lt;/p&gt;

  </p>
</div>
`;

exports[`should render a list of comments 1`] = `
Array [
  <div>
    <div>
      Chris Ames
    </div>
    <p>
      &lt;p&gt;Ello! Pretend you&#8217;re reading this comment with an English accent.&lt;/p&gt;

    </p>
  </div>,
  <div>
    <div>
      Jared Erickson
    </div>
    <p>
      &lt;p&gt;Non-breaking text is my favorite!&lt;/p&gt;
&lt;p&gt;Super/Duper/Long/NonBreaking/Path/Name/To/A/File/That/Is/Way/Deep/Down/In/Some/Mysterious/Remote/Desolate/Part/Of/The/Operating/System/To/A/File/That/Just/So/Happens/To/Be/Strangely/Named/Supercalifragilisticexpialidocious.txt&lt;/p&gt;

    </p>
  </div>,
]
`;
